### PR TITLE
resources/images: Create AutoOrient image filter

### LINKED ISSUE #11717: Create AutoOrient image filter
This would allow users to fix orientation issues when `.Exif.Tags.Orientation` is one of 2, 4, 5 or 7.

Ref: <https://discourse.gohugo.io/t/image-processing-rotates-images/>

### PR DESCRIPTION
**Pull request by jmooring**
Closes #11717

### CODE REVIEW

**Comment by jmooring** [comment]
To test:

```text
git clone --single-branch -b hugo-github-issue-11717 https://github.com/jmooring/hugo-testing hugo-github-issue-11717
cd hugo-github-issue-11717
hugo server
```

I was unable to change the length of the filters slice without errors, nor was I able to return a nil filter. Returning a noop filter seemed reasonable. The gift package does not provide a noop filter, but this is effectively a noop:
```
// Return a noop filter.
return gift.Brightness(0)
```

**Review by bep — CHANGES_REQUESTED** [review]

File: resources/images/auto_orient.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		5: gift.Transpose(),
+		6: gift.Rotate270(),
+		7: gift.Transverse(),
+		8: gift.Rotate90(),
+	}
+
+	if exifInfo != nil {
+		orientation, ok := exifInfo.Tags["Orientation"].(int)
+		if ok && orientation >= 2 && orientation <= 8 {
+			return filters[orientation]
```
[the remark this answers is not in the document]
Comment (jmooring) [reply resources/images/auto_orient.go]: Done.

File: resources/images/auto_orient.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		}
+	}
+
+	// Return a noop filter.
+	return gift.Brightness(0)
```
[the remark this answers is not in the document]
Comment (jmooring) [reply resources/images/auto_orient.go:60]: > returned nil here and removed it from the filter slice on the receiving end

That was my first approach, but reducing the number of elements in the filter slice caused errors elsewhere:

>  executing "index.html" at <images.Filter>: error calling Filter: runtime error: slice bounds out of range [2:1]

I try to track that down when I have some more time.

[the remark this answers is not in the document]
Comment (jmooring) [reply resources/images/auto_orient.go]: Done

**Pull request merged**